Sources/TootSDK/TootClient/TootClient+ASWebAuthentication.swift

```diff
@@ -0,0 +1,62 @@
+// Created by david-hosier on 07/02/2023
+// Copyright (c) 2023. All rights reserved.
+
+#if !os(Linux)
+
+import AuthenticationServices
+import Foundation
+
+extension TootClient {
+    
+    /// Presents a authentication/authorization page using ASWebAuthenticationSession.
+    /// When signIn is initiated on supported platforms, the authorization web page will be presented in a system-dependent manner that negates the need for user's to implement a custom web view. Upon successful authentication (or cancellation), control will be return to this function to handle token retrieval.
+    ///
+    /// - Parameters:
+    ///   - callbackURI: The callback URI  (`redirect_uri`) which was used to initiate the authorization flow. Must match one of the redirect_uris declared during app registration.
+    ///   - prefersEphemeralWebBrowserSession: A Boolean value that indicates whether the session should ask the browser for a private authentication session.
+    ///   - presentationContextProvider: A delegate that provides a display context in which the system can present an authentication session to the user.
+    /// - Returns: The auth token for the user if authentication succeeds.
+    @MainActor public func presentSignIn(callbackURI: String,
+                                         prefersEphemeralWebBrowserSession: Bool = true,
+                                         presentationContextProvider: ASWebAuthenticationPresentationContextProviding? = TootPresentationAnchor()) async throws -> String {
```

Comment:
```suggestion
                                         presentationContextProvider: ASWebAuthenticationPresentationContextProviding = TootPresentationAnchor()) async throws -> String {
```

Non-optional so it will be impossible to call this incorrectly. Unless there is a case where nil is valid, but I’m not aware of one.